src/bin/pg_autoctl/pgsetup.c

```diff
@@ -411,6 +411,16 @@ get_pgpid(PostgresSetup *pgSetup, bool pg_is_not_running_is_ok)
 
 	join_path_components(pidfile, pgSetup->pgdata, "postmaster.pid");
 
+	if (!file_exists(pidfile))
```

Comment:
is it because we want to have a different error message ? I suppose `read_file` returns false when file does not exist

Reply:
It's because `read_file` always outputs an ERROR line when the file does not exists, and the `get_pgpid` might just be probing if the file exists or not, and it's very fine for it to not exists... so that we don't want an ERROR message to be unconditionally part of the output in that case.